units/stats_task_end.go

```diff
@@ -151,6 +151,10 @@ func (j *collectTaskEndDataJob) Run(ctx context.Context) {
 		msg["dependencies_met_time"] = j.task.DependenciesMetTime
 	}
 
+	if !j.task.ContainerAllocatedTime.IsZero() {
```

Comment:
If you use Unix zero time as the zero value, this will only work if you use `utility.IsZeroTime` because `IsZero` only checks for Go zero time.